### PR TITLE
chore: Keep CLI minimum-version reminders accurate

### LINKED ISSUE #1222: Move CLI minimum warning comment logic to Go
## Context

PR #1220 moved the CLI minimum version warning regression coverage into Go tests, but `scripts/comment-cli-minimum-version-warning.sh` still owns the warning decision logic and GitHub comment upsert/resolve behavior.

The shell script is now harder to maintain than the tests around it. Moving the implementation behind a small Go entrypoint should make the behavior easier to test, reduce shell-specific edge cases, and keep the tests close to the code they protect.

## Scope

- Move the decision logic from `scripts/comment-cli-minimum-version-warning.sh` into Go.
- Move GitHub warning comment find/create/update/resolve behavior into Go.
- Keep the shell script as a thin GitHub Actions entrypoint.
- Preserve the current external behavior of the script and workflow.
- Keep the implementation small; do not combine this with GitHub Actions SHA pinning or workflow security policy changes.

## Acceptance criteria

- The shell entrypoint delegates to a Go implementation instead of owning the warning decision logic.
- GitHub comment find/create/update/resolve behavior is covered by focused Go tests.
- Existing CLI minimum version warning scenarios continue to pass.
- The GitHub Actions workflow keeps the same external behavior after the implementation move.
- `scripts/check-go-cli.sh` passes.

## Related

- Split out from #1221.
- Follow-up from PR #1220 review and discussion.

### PR DESCRIPTION
**Pull request by hatayama**
## Summary
- Pull request authors now get the CLI minimum-version reminder from the same Go toolchain that owns native CLI validation.
- The reminder ignores automation-only Go files and keeps successful GitHub CLI notices out of parsed command output.

## User Impact
- PRs that change shipped native CLI behavior still get a clear reminder to confirm `MINIMUM_REQUIRED_CLI_VERSION`.
- PRs that only update the reminder automation no longer get misleading compatibility warnings.

## Changes
- Added a small Go command for CLI minimum-version comment detection and GitHub comment upsert/resolve.
- Reduced the shell script to a GitHub Actions entrypoint and set up Go in the workflow.
- Added focused tests for diff classification, stdout parsing, and workflow behavior.

## Verification
- `go test ./internal/automation ./internal/architecture -run TestMinimumVersion -count=1`
- `scripts/test-comment-cli-minimum-version-warning.sh`
- `scripts/test-cli-minimum-version-warning-workflow.sh`
- `git diff --check`
- `scripts/check-go-cli.sh`
- `~/.codex/skills/codex-review/scripts/codex-review --parallel-tests "scripts/check-go-cli.sh" v3-beta`

Closes #1222

### CODE REVIEW

File: Packages/src/Cli~/internal/automation/minimum_version_warning.go [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+	if err != nil {
+		writeMinimumVersionWarningLine(stderr, err)
+		return 1
+	}
+	config.repository = repository
+
+	if config.baseRef == "" {
+		writeMinimumVersionWarningLine(stdout, "Skipping CLI minimum version comment because no base ref was provided.")
+		return 0
+	}
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Cli~/internal/automation/minimum_version_warning.go]: Fixed in 621e2292.

I moved the missing base-ref skip before repository resolution, so this path no longer shells out to `gh` when it should exit cleanly. I also added a regression test for that behavior.

File: Packages/src/Cli~/internal/automation/minimum_version_warning.go [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+	if err != nil {
+		writeMinimumVersionWarningLine(stderr, err)
+		return 1
+	}
+	if config.pullRequest == "" {
+		writeMinimumVersionWarningLine(stdout, "Skipping CLI minimum version comment because no PR number was provided.")
+		return 0
+	}
+
+	repository, err := resolveMinimumVersionWarningRepository(ctx, config)
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Cli~/internal/automation/minimum_version_warning.go:53]: Fixed in 621e2292.

I moved the missing base-ref skip before repository resolution, so this path no longer shells out to `gh` when it should exit cleanly. I also added a regression test for that behavior.

File: Packages/src/Cli~/internal/automation/minimum_version_warning.go [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+	switch changedFile {
+	case goCliPackageRoot + "go.mod",
+		goCliPackageRoot + "go.sum",
+		goCliPackageRoot + "contract.json",
+		goCliPackageRoot + "layout-contract.json":
+		return true
+	default:
+		relativePath := strings.TrimPrefix(changedFile, goCliPackageRoot)
+		return relativePath != changedFile && strings.HasSuffix(relativePath, ".go") && !strings.Contains(relativePath, "/")
+	}
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Cli~/internal/automation/minimum_version_warning.go:251]: Fixed in 621e2292.

I excluded `*_test.go` files from shipped native CLI change detection and added a regression case for `internal/architecture/comment_cli_minimum_version_warning_test.go`.

File: Packages/src/Cli~/internal/automation/minimum_version_warning.go [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+		return "", fmt.Errorf("%s %s failed: %w\n%s%s", name, strings.Join(args, " "), err, stderr.String(), stdout.String())
+	}
+	return stdout.String(), nil
+}
+
+func minimumVersionWarningIsGoCliFile(changedFile string) bool {
+	if changedFile == goCliPackageRoot+"CHANGELOG.md" ||
+		strings.HasPrefix(changedFile, goCliPackageRoot+"dist/") ||
+		strings.HasPrefix(changedFile, goCliPackageRoot+"cmd/comment-cli-minimum-version-warning/") ||
+		strings.HasPrefix(changedFile, goCliPackageRoot+"internal/automation/") {
```
[the remark this answers is not in the document]
Comment (hatayama) [reply Packages/src/Cli~/internal/automation/minimum_version_warning.go]: Fixed in 621e2292.

I excluded `*_test.go` files from shipped native CLI change detection and added a regression case for `internal/architecture/comment_cli_minimum_version_warning_test.go`.
